transport/ice/IceSession.cpp

```diff
@@ -1091,31 +1270,172 @@ void IceSession::stateCheck(const uint64_t now)
 
 void IceSession::freezePendingProbes(uint64_t now)
 {
-    for (auto it = _candidatePairs.rbegin(); it != _candidatePairs.rend(); ++it)
+    // We are going to clear all remote candidates without candidatePairs
+    _remoteCandidates.clear();
```

Comment:
There is a bit unsafe if it happens before all candidate pairs have been created because probes are added at pace. Remote candidates towards end of list will be lost.

Reply:
Candidate pairs are created right after creating PRFX candidate. We are only removing PRFLX candidates, meaning that we are not deleting anything does not have probes (exception is for the TCP canidates that we don't care if they are PRFLX as we are agressive on prune tcp connection).

But yes, we could let the probes to complete and delete the failed ones, so we keep only the candidates that work. I will change that.

Regarding TCP, transport is already deleting all TCP endpoints then this logic of keeping TCP that does not have valid probes on UDP is not needed as transport will kill the socket.

Reply:
I rename this method to not crate confusing. It is now `pruneCandidatesAfterConnected` (and it should be called once the states gets connected).

This removes:
 - Failed candidates (regardless if they are PRFLX, SRFLX, HOST, etc)
 - TCP candidates. TransportImpl will delete the endpoints anyway
 
 This freezes the Succeeded candidates if not nominated (this will make SMB stops to probe them). Its completely to the client to keep them 'viable' by continue probing them (I guess C9 is much more aggressive on freeing candidates so let the clients to decide) 
 
 It does not touch on InProgress candidate pairs. They must Fail or Succeed  at some point. When a candidate pair Succeed after state is CONNECTED, the state transits to frozen instead.
 
I changed the method `processTimeout` to call `removeUnviableRemoteCandidates` when the last InProgress candidate completes its state and ICE state is CONNECTED (so we can remove the unviable candidates in the end without run the method each time a candidate fails)